Modules/Filtering/ImageIntensity/include/itkHistogramMatchingImageFilter.hxx

```diff
@@ -277,8 +343,8 @@ void
 HistogramMatchingImageFilter< TInputImage, TOutputImage, THistogramMeasurement >
 ::DynamicThreadedGenerateData(const OutputImageRegionType & outputRegionForThread)
 {
-
-  InputImageConstPointer input  = this->GetInput();
+  constexpr unsigned int source_image_idx=0;
+  InputImageConstPointer input  = this->GetInput(source_image_idx);
```

Comment:
`GetInput(0)` would have worked too, and had one less entity (`source_image_idx`) to contend with.

Reply:
@dzenanz It used to be GetInput(0), but I was trying to remove "magic numbers".  The constexpr could also have been `#define soruce_image_idx=0`.  I was trying to increase readability, and provide a token to search for in the code so that I could isolate all the places where source_image_idx was used.  Searching for `0` did not provide a useful query :).

After addressing comments by Brad, I'm only using named inputs, so this is now not relevant.